azure_img_utils/azure_image.py

```diff
@@ -96,6 +96,11 @@ def __init__(
 
     def image_blob_exists(self, blob_name: str):
         """Return True if image blob exists in the configured container."""
+        if not self.container:
```

Comment:
so Im not as familiar to this codebase as you both are, in [this line](https://github.com/SUSE-Enceladus/azure-img-utils/blob/main/azure_img_utils/storage.py#L46)  code returns True|False if blob exists
which is the method invoked in [here](https://github.com/SUSE-Enceladus/azure-img-utils/blob/main/azure_img_utils/azure_image.py#L97)
in that `image_blob_exists` you are adding an if not `None` check for the container in the PR

I have been testing it (after some fiddling with the tests) and if container is None or whatever that does not exist, it returns false but outside of this test envionment of `azure-img-utils`, a non existing container would return AuthenticationFailed error
when calling a non existing container (when authenticating in this [line](https://github.com/SUSE-Enceladus/azure-img-utils/blob/main/azure_img_utils/storage.py#L54))
```python3
blob_client = blob_service_client.get_container_client(container="foo")
# exception:
Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.
...
```
and for a empty container it would return 
```python3
blob_client = blob_service_client.get_container_client(container=None)
# exception:
Please specify a container name and blob name
```

There is another option/change available (I guess depending on the version ?) which is to change these [lines](https://github.com/SUSE-Enceladus/azure-img-utils/blob/main/azure_img_utils/storage.py#L113-L114)  for:
```python
return blob_service_client.get_blob_client(container=container_name, blob=blob_name)
```
This would have a different exception if container name is populated but not existing
```python
blob_client = blob_service_client.get_blob_client(container="foo", blob=blob)
# exception:
Operation returned an invalid status 'Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.'
ErrorCode:AuthenticationFailed
```
and also it would return the same when container name is `None` for a empty container, this is
```python3
blob_client = blob_service_client.get_blob_client(container=None, blob=blob)
# exception:
Please specify a container name and blob name
```

so maybe, in any of the options (`get_container_client` or `get_blob_client(container, blob)`)
```python3
try:
    return blob_exists(
        self.blob_service_client,
        blob_name,
        self.container
    )
except Exception as err:
    raise AzureImgUtilsException(err)
```
As I do not see the case of a container value being populated but non existing in the storage account,
that would cover both cases

Thoughts ?

Reply:
The main goal was to not require args at class instantiation that might not be needed. Depending on the subsequent actions. But with container and storage account not required at class creation the user could end up with confusing auth errors from Azure. Thus catching these early makes the message clear.

However, as you pointed out in the continer and blob client case Azure actually provides the same usable ValueError. So I amended the PR to catch this and re-raise. Which prevents duplicating the check in the class.